src/device/rdk/system/settings/set.rs

```diff
@@ -245,7 +228,8 @@ fn set_rdk_text_to_speech(val: bool) -> Result<(), String> {
     Ok(())
 }
 
-pub fn process(_packet: String) -> Result<String, String> {
+pub fn process(_dab_request: SetSystemSettingsRequest) -> Result<String, DabError> {
```

Comment:
Because you're passing whole `SystemSettingsRequest` structure instead of serialized JSON request, `process()` will now not only change the settings to the values from the DAB request, but will also attempt to set **all** other settings (the ones DAB request did not change) to their default values.

See https://github.com/device-automation-bus/dab-adapter-rs/commit/7b6d09cf8e46295fbe90cd003c82c3ecd54d97c2 for my suggested fix.